### PR TITLE
Next attempt to enable GDML for Geant4

### PR DESCRIPTION
**Pull request by sawenzel**
* build G4 with GDML support
* update recipe for XercesC with correct tag
* use tools to generate modulefiles for XercesC + Geant4 (code reduction)

Note that in the past we had problems in relocating XercesC + Geant4. Let's hope that this has been fixed meanwhile.

### CODE REVIEW

**Comment by sawenzel** [comment]
@TimoWilken : We need to cross-check that it still builds whenever XercesC is coming from the remote store. In past, this wasn't the case.

File: geant4.sh [hunk 1]
```diff
@@ -56,22 +60,12 @@ make install
...
-set version $PKGVERSION-@@PKGREVISION@$PKGHASH@@
-module-whatis "ALICE Modulefile for $PKGNAME $PKGVERSION-@@PKGREVISION@$PKGHASH@@"
-# Dependencies
-module load BASE/1.0 ${XERCESC_REVISION:+xercesc/$XERCESC_REVISION-$XERCESC_REVISION}
-# Our environment
+alibuild-generate-module --bin --lib > $MODULEFILE
+cat >> "$MODULEFILE" <<EOF
+# extra environment
 set GEANT4_ROOT \$::env(BASEDIR)/$PKGNAME/\$version
 setenv GEANT4_ROOT \$GEANT4_ROOT
```
Comment (ktf) [inline geant4.sh:67]: Do we really need GEANT4_ROOT?

Comment (sawenzel) [reply geant4.sh:67]: It was there before so I kept it. It cannot hurt and is independent of the intention of this PR.

File: xercesc.sh [hunk 2]
```diff
@@ -27,17 +28,8 @@ make install
...
-}
-set version $PKGVERSION-@@PKGREVISION@$PKGHASH@@
-module-whatis "ALICE Modulefile for $PKGNAME $PKGVERSION-@@PKGREVISION@$PKGHASH@@"
-# Dependencies
-module load BASE/1.0
-# Our environment
+alibuild-generate-module --lib > $MODULEFILE
+cat >> "$MODULEFILE" <<EOF
+# extra environment
 set XERCESC_ROOT \$::env(BASEDIR)/$PKGNAME/\$version
```
[the remark this answers is not in the document]
Comment (TimoWilken) [reply xercesc.sh:34]: This can definitely be removed -- it only sets a Tcl variable, so it'll be set and thrown away immediately.

**Comment by TimoWilken** [comment]
> @TimoWilken : We need to cross-check that it still builds whenever XercesC is coming from the remote store. In past, this wasn't the case.

I can build this and upload it to the remote store, then try reusing it. What's a simple command I can run in the alienv environment to make sure it's working?

**Pull request merged**